client/mysqlbinlog.cc

```diff
@@ -2699,7 +2701,10 @@ static Exit_status dump_remote_log_entries(PRINT_EVENT_INFO *print_event_info,
       File file;
 
       if ((file= load_processor.prepare_new_file_for_old_format(le,fname)) < 0)
+      {
+        reset_temp_buf_and_delete(ev);
         DBUG_RETURN(ERROR_STOP);
```

Comment:
Sorry, which one DBUG_RETURN(retval) is OK without reset_temp_buf_and_delete(ev)? Line 2692 is neither